src/project/python.rs

```diff
@@ -25,6 +26,22 @@ pub enum ParsePyPiRequirementError {
     MissingOperator(String),
 }
 
+impl fmt::Display for PyPiRequirement {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let all_extras = self
+            .extras
+            .clone()
+            .map(|each_str| each_str.join(", "))
+            .unwrap_or_default();
+        write!(
+            f,
+            "version=\"{}\", extras=\"{}\"",
+            self.version.clone().unwrap(),
```

Comment:
It good practice to never have an `.unwrap()` in production code. You normally want to catch the error and return it or modify it slightly to help the user understand what is going on. When in theory you can never reach that code in production you can use an `.expect("message")` to help the developer find the issue faster by making the message say something about why you should never reach that code.

Reply:
Also the `clone` is unnecessary, probably `as_ref` works too.

Reply:
This bit was reformulated